### PR TITLE
migrate pipeline processor and filter tests to rspec

### PR DESCRIPTION
**Pull request by ericmustin**
This PR moves over the `Datadog::Pipeline`, `Datadog::Pipeline::SpanProcessor`, and `Datadog::Pipeline::SpanFilter` minitest tests within the /tests folder to rspec spec's within the /spec folder. It tests what was covered in the minitest suite, specifically, the behavior of the trace pipeline, and the functionality of both OOTB processor and filter classes as well as arbitrary processors or filters as described here: https://github.com/DataDog/dd-trace-rb/blob/master/docs/GettingStarted.md#processing-pipeline

### CODE REVIEW

File: spec/ddtrace/pipeline_spec.rb [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  let(:span_b) { generate_span('b') }
+  let(:span_c) { generate_span('c') }
+  let(:span_d) { generate_span('d') }
+  let(:span_list) { [span_a, span_b, span_c, span_d] }
+  let(:callable) { ->(trace) { trace } }
+  let(:custom_callable_conditional) { ->(trace) { trace if trace.size == 3 } }
+  let(:custom_callable_reverse) { ->(trace) { trace.reverse } }
+
+  before do
+  end
```
Comment (marcotc) [inline spec/ddtrace/pipeline_spec.rb:22]:
```suggestion
```

File: spec/ddtrace/pipeline/span_filter_spec.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+require 'ddtrace/pipeline'
+require 'ddtrace/span'
+
+RSpec.describe Datadog::Pipeline::SpanFilter do
+  def generate_span(name, parent = nil)
+    Datadog::Span.new(nil, name).tap do |span|
+      span.parent = parent
+    end
+  end
+
```
Comment (marcotc) [inline spec/ddtrace/pipeline/span_filter_spec.rb:11]: You might want to move this method in to a shared file, like Sidekiq does here: https://github.com/DataDog/dd-trace-rb/tree/64ec7458bd9e6fe010382cdd655044a8aca7264f/spec/ddtrace/contrib/sidekiq/support

We normally add a `support/` directory with shared files in it.

File: spec/ddtrace/pipeline_spec.rb (hunk 1 above)
Comment (ericmustin) [reply spec/ddtrace/pipeline_spec.rb]: 👍  updated

File: spec/ddtrace/pipeline/span_filter_spec.rb (hunk 2 above)
Comment (ericmustin) [reply spec/ddtrace/pipeline/span_filter_spec.rb]: makes sense, updated

**Pull request merged**